src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer_ILspLogger.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.LogHub;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
+{
+    internal partial class InProcLanguageServer : ILspLogger
+    {
+        public async Task<TraceSource?> CreateTraceSourceAsync(string logName, CancellationToken cancellationToken)
+        {
+            if (_asyncServiceProvider == null)
+                return null;
+
+            var cleaned = string.Concat(logName.Split(Path.GetInvalidFileNameChars(), StringSplitOptions.RemoveEmptyEntries));
+            var logId = new LogId(cleaned, new ServiceMoniker("Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient"));
+
+            var serviceContainer = await _asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+            var service = serviceContainer.GetFullAccessServiceBroker();
+
+            using var configuration = await TraceConfiguration.CreateTraceConfigurationInstanceAsync(service, cancellationToken).ConfigureAwait(false);
```

Comment:
You're on really shaky ground here with regard to the lifetime of `TraceConfiguration`. See [this bug](https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1258152) I filed against LogHub for being all over the map here.
I have no idea what to recommend you do here, but you could try asking the Extensibility team for clarity and/or to fix the bug.

Reply:
i'll keep the lifetimes of these two together.